src/clj/clojure/lang/map_entry.clj

```diff
@@ -20,6 +33,41 @@
     (list*
       'deftype type ['k 'v]
 
+      'ICounted
+      (list '-count ['this] 2)
+
+      'IIndexed
+      (list '-nth ['this 'n]
+        (list 'cond
+          (list 'zero? 'n) 'k
+          (list '= 'n 1) 'v
+          :else (list 'throw (list 'new-out-of-bounds-exception ""))))
+
+      (list '-nth ['this 'n 'not-found]
+        (list 'if (list 'or (list '= 'n 0) (list '= 'n 1))
+          (list '-nth 'this 'n)
+          'not-found))
+
+      'ILookup
+      (list '-lookup ['this 'k 'not-found]
+        (list 'if (list 'is-integer? 'k)
```

Comment:
We do an int cast on the line below which will blow up unless it's one of the integer types and we want it to return the `not-found`

https://github.com/clojure/clojure/blob/clojure-1.6.0/src/jvm/clojure/lang/APersistentVector.java#L339